src/js/msp/MSPHelper.js

```diff
@@ -1620,20 +1619,22 @@ MspHelper.prototype.process_data = function(dataHandler) {
         }
     }
     // trigger callbacks, cleanup/remove callback after trigger
-    for (let i = dataHandler.callbacks.length - 1; i >= 0; i--) { // itterating in reverse because we use .splice which modifies array length
-        if (dataHandler.callbacks[i].code == code) {
-            // save callback reference
-            var callback = dataHandler.callbacks[i].callback;
-            var callbackOnError = dataHandler.callbacks[i].callbackOnError;
-
-            // remove timeout
-            clearInterval(dataHandler.callbacks[i].timer);
-
-            // remove object from array
-            dataHandler.callbacks.splice(i, 1);
-            if (!crcError || callbackOnError) {
-                // fire callback
-                if (callback) callback({'command': code, 'data': data, 'length': data.byteLength, 'crcError': crcError});
+    for (let i = dataHandler.callbacks.length - 1; i >= 0; i--) { // iterating in reverse because we use .splice which modifies array length
+        if (dataHandler.callbacks[i] !== undefined) {
+            if (dataHandler.callbacks[i]?.code === code) {
```

Comment:
In this case (`?.`) the outer `if (dataHandler.callbacks[i] !== undefined) {` is not needed anymore, as `dataHandler.callbacks[i]?.code` will yield `undefined` if `dataHandler.callbacks[i]` does not exist. :-)

Reply:
Err, one Control-Z to many. Fixed